cl/free_look_email/app.py

```diff
@@ -0,0 +1,88 @@
+import json
+import requests
+from humps import decamelize
+import random, time
+
+def retry_with_backoff(fn, retries = 10, backoff_in_seconds = 5):
+  x = 0
+  while True:
+    try:
+      return fn()
+    except:
+      if x == retries-1:
+        raise
+      else:
+        sleep = (backoff_in_seconds * 2 ** x +
+                 random.uniform(0, 1))
+        time.sleep(sleep)
+        x += 1
+
+def get_email_record_from_event(event):
+    return decamelize(event['Records'][0]['ses']['mail'])
+
+def get_email_receipt_from_event(event):
+    return decamelize(event['Records'][0]['ses']['receipt'])
+
+def get_spam_verdict(receipt):
+    return receipt['spam_verdict']['status']
+
+def get_virus_verdict(receipt):
+    return receipt['virus_verdict']['status']
+
+def get_spf_verdict(receipt):
+    return receipt['spf_verdict']['status']
+
+def get_dkim_verdict(receipt):
+    return receipt['dkim_verdict']['status']
+
+def get_dmarc_verdict(receipt):
+    return receipt['dmarc_verdict']['status']
+
+def get_combined_log_message(email):
+    subject = None
+    for header in email['headers']:
+        if header['name'] == 'Subject':
+            subject = header['value']
+
+    return 'Email with {subject}, from {source} to {destination}'.format(subject=subject, source=email['source'], destination=email['destination'])
+
+def validation_failure(email, receipt, verdict):
+    print('{combined} failed with spam verdict {verdict}'.format(
+        combined=get_combined_log_message(email),
+        verdict=verdict))
+    return {
+        "statusCode": 424,
+        "body": json.dumps(receipt),
+    }
+
+def send_to_court_listener(email, receipt):
+    # TODO - Docker internal host here should be an environment variable.
+    court_listener_response = requests.post(
```

Comment:
@mlissner Regarding your main question, this is where it hits the CourtListener API. In this service, I don't plan to have it do anything except email verification and the retry. Everything else gets sent over to CourtListener.

Not sure if that quite answers it, but we can clarify on Slack via a longer conversation if you like.